frmts/vrt/vrtderivedrasterband.cpp

```diff
@@ -108,15 +117,14 @@ static PyObject *GDALCreateNumpyArray(PyObject *pCreateArray, void *pBuffer,
             break;
         case GDT_CInt16:
         case GDT_CInt32:
-            CPLAssert(FALSE);
-            break;
```

Comment:
OK to coerce these types to CFloat32?

Reply:
I don't remember the details of how this works, but it is not obvious to me that this will cause a cint16/cint32 -> cfloat32 type conversion. This might require a prior type conversion
Actually I see at lines 1208 - 1220 that we already error out on GDT_CInt16/GDT_CInt32, so the assertion here is legit

Reply:
restored it